### PR TITLE
OEP-18 compliance

### PR DESCRIPTION
**Pull request by timmc-edx**
This PR is in two parts: One commit to migrate to a `make upgrade` world but with no change in versions, and a second commit to show the change in versions when the upgrade actually happens (and to fix the test runner, which is broken for the intermediate commit.)

### CODE REVIEW

File: requirements/base.in [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Core requirements for using this package template
+-c constraints.txt
+
+Django
+PyJWT
+six
+social-auth-app-django
+social-auth-core
```
Comment (jmbowman) [inline requirements/base.in]: You could omit this one, social-auth-app-django declares it as a dependency so it'll get pulled in anyway and I don't think we use it directly.

File: requirements/base.in [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Core requirements for using this package template
+-c constraints.txt
+
+Django
```
Comment (jmbowman) [inline requirements/base.in]: Maybe preserve the lower bounds from the old `setup.py` in here?

File: requirements/base.in (hunk 1 above)
Comment (timmc-edx) [reply requirements/base.in]: It looks like `auth_backends/strategies.py` might use it directly (`social_core` around lines 20–30)—does that change anything in your estimation?

Comment (jmbowman) [reply requirements/base.in]: Sure, go ahead and leave it with a comment at the end of the line explaining briefly what we directly use it for.

**Pull request merged**